### PR TITLE
Implement nanvar function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanvar, which computes the variance of elements along a specified axis while ignoring NaN values in the input tensor.
The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -5432,6 +5432,56 @@ def nansum(x, axis=None, keepdims=False):
...
+
+
+@keras_export(["keras.ops.nanvar", "keras.ops.numpy.nanvar"])
+def nanvar(x, axis=None, keepdims=False):
+    """Compute the variance along the specified axes, ignoring NaNs.
+
+    Args:
+        x: Input tensor.
+        axis: Axis or axes along which the variance is computed.
+        keepdims: If True, reduced axes are kept as size-1 dims.
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py]: updated.

**Review by hertschuh — APPROVED** [review]
Thank you for adding this!
